### PR TITLE
GSOC-DS-005 Refactor project loading

### PR DESCRIPTION
**Pull request by DinosaurierRex**
- Moved all project loading specific tasks to new class
- Tried to avoid using context that much (Interface and implementation of a Stringprovider for Project loading)
- Refactor of test and code usages

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by gPathpp — CHANGES_REQUESTED** [review]

**Comment by DinosaurierRex** [comment]
@gPathpp Requested changes resolved. Tests which fail on Jenkins run locally without Problems. May this is still related to slow emulators for test runs.

File: catroid/src/main/java/org/catrobat/catroid/io/ProjectLoadAndUpdate.kt [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+                } catch (e: IOException) {
+                    iterator.remove()
+                }
+            }
+        }
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun updateScriptsToTreeStructure(project: Project) {
+        for (scene in project.sceneList) {
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/io/ProjectLoadAndUpdate.kt]: Please use Project Updater

File: catroid/src/main/java/org/catrobat/catroid/io/ProjectLoadAndUpdate.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+            val objectFile = when (objectData) {
+                is LookData -> objectData.file
+                is SoundInfo -> objectData.file
+                else -> null
+            }
+
+            if (existingFiles.contains(objectFile?.name)) {
+                try {
+                    if (objectData is LookData) objectData.file = duplicateFile(objectData.file)
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/io/ProjectLoadAndUpdate.kt:234]: As discussed, please use callbacks instead

**Review by gPathpp — CHANGES_REQUESTED** [review]
Looks good!

**Review by gPathpp — APPROVED** [review]
Well done 👍

**Pull request closed without merging**